### DISCUSSION TITLE
Problems using 256GB SD Card [disk space is not being fully used]

### DISCUSSION THREAD

**Discussion by tigers1995** [iiab/iiab#3339] [Q&A] [answered]
I purchased 2 new Sandisk Ultra 256GB sd cards and I installed the IIAB image using the Raspberry Pi Imager. For some reason 1 only has 15 GB of available storage and the other has 60 GB of storage available before I installed any content.  Does anyone know why that might be happening?  Thanks in advance.

**Comment by holta** [comment]
@tigers1995 thanks for writing and apologies for the hassle!

Do you have command-line access on Linux?  With an attached screen/keyboard or via SSH ?

If so, the short-term fix is to run:

1. ```sudo iiab-expand-rootfs```
2. Wait patiently (for about 3 minutes) until the disk expands to the full capacity of the microSD card.
   (You can monitor this progress using the `df -h` command.)

Does that work for you?

Just FYI, here is some ongoing work to fix this properly:

- #3325
- PR #3337

**Comment by tigers1995** [comment]
I do not have linux. I used a PC to install the IIAB zip file. The SD card
is installed in a Raspberry Pi 400 with keyboard and I have an ethernet
connection.

**Comment by holta** [comment]
Jeremy and Jacqueline,

Do you have a screen attached to your Raspberry Pi 400?

If so you have access to Linux, so you can run the aforementioned command: `sudo iiab-expand-rootfs`

Let me know!  (Or we can do a quick video call if nec...)

Adam
holt@unleashkids.org

**Comment by holta** [comment]
> Do you have a screen attached to your Raspberry Pi 400?
>
> If so you have access to Linux, so you can run the aforementioned command: sudo iiab-expand-rootfs

Any progress Jeremy & Jacqueline?

Let us know!  Or if a quick call's nec :smile:

Adam
holt@unleashkids.org

**Comment by tigers1995** [comment]
I tried the command line you mentioned and then I was asked for additional
parameters I believe and I was not sure what to pick. Thank you so much for your quick replies

**Comment by tigers1995** [comment]
I am not sure how to modify the command. I am new to the pi and I appreciate your help. I tried google but non of the stuff I found seemed to fit what I need with iiab

![image](https://user-images.githubusercontent.com/110906020/183926649-dfbaba79-9c94-4500-91e8-82eafab85431.png)

**Reply by jvonau** [reply]
@holta Should you do a `sudo touch /.expand-rootfs ` before calling that routine? Otherwise there is an exit without that file present, see full [rountine](https://github.com/iiab/iiab/issues/3325#issuecomment-1199259094)

**Comment by holta** [comment]
@tigers1995 how long did the command take to run approximately?

Did you then run `df -h` to see if it (hopefully!?) worked to expand your "disk" space up to 200+ GB ?

**Reply by tigers1995** [reply]
fraction of a second to run and came back with the 2 lines with the plus signs. Dev/root on this SD card is still 61 GB

**Reply by holta** [reply]
Ok!  Any chance you can call to help make this more efficient?

Adam
holt@unleashkids.org

**Comment by tigers1995** [comment, marked as answer]
@jvonau , that did the trick. I ran  sudo touch /.expand-rootfs and then
I ran  sudo
iiab-expand-rootfs.

Now the space available is 222 GB. Thanks all.
